src/system/kernel_linux/Team.cpp

```diff
@@ -30,16 +47,80 @@ _kern_kill_team(team_id team)
 status_t
 _kern_get_team_info(team_id id, team_info *info)
 {
-	info->team = id;
-	return B_OK;
+  	char proc_path[500];
+    char buffer[MAX_LINE_LENGTH];
+    char command_proc_path[500];
+    sprintf(proc_path, "/proc/%d/status", id);
+    sprintf(command_proc_path, "/proc/%d/cmdline", id);
+  	FILE *fileid = fopen(proc_path, "r");
+    int file_command_id = open(command_proc_path, O_RDONLY);
+    if( fileid == NULL ) {
+		return B_ERROR;
+	}
+    int i=0;
+    while( !feof(fileid) ){
+        fgets(buffer, MAX_LINE_LENGTH, fileid);
+        if( i == UID_LINE ) {
+            sscanf(buffer, "Uid:\t%u\t", &info->uid);
+        } else if( i == GID_LINE ) {
+            sscanf(buffer, "Gid:\t%u\t", &info->gid);
+        } else {
+            if(strncmp(buffer, "Threads:", sizeof("Threads:")) == 0){
+                sscanf(buffer, "Threads:\t%d", &info->thread_count);
+                break;
+            }
+        }
+        i++;
+    }
+    int bytes_read = read(file_command_id, buffer, MAX_LINE_LENGTH);
+    int argCount = 0;
+    int j = 0;
+    while( j < bytes_read ){
+        if( buffer[j] == '\0' ){
+            argCount++;
+            buffer[j] = ' ';
+        }
+        j++;
+    }
+    buffer[j] = '\0';
+    info->argc = argCount;
+    strncpy(info->args, buffer, MAX_BYTES);
+    fclose(fileid);
+    close(file_command_id);
+    return B_OK;
 }
 
 
 status_t
 _kern_get_next_team_info(int32 *cookie, team_info *info)
 {
-	UNIMPLEMENTED();
-	return B_ERROR;
+    if(*cookie == 0){
+        DIR *d;
+        struct dirent *dir;
+        
+        d = opendir("/proc");
+        
+        if(d){
+            while ((dir = readdir(d)) != NULL){
+                if(atoi(dir->d_name) > 0) {
+                    gTeams.push_front(atoi(dir->d_name));
+                }
+            }
+        } else {
+            return B_ERROR;
+        }
+        closedir(d);
+    }
+    if(*cookie > gTeams.size()){
+        return B_ERROR;
+    }
+    list <int> :: iterator gTeamsIt = gTeams.begin();
+    std::advance(gTeamsIt, *cookie);
+    unsigned int cur_pid = *gTeamsIt;
+    status_t retval = _kern_get_team_info(cur_pid, info);
```

Comment:
no need to declare retVal, increase the cookie and then return directly _kern_get_team_info